### PR TITLE
Don't act differently on 0 offset M5

### LINKED ISSUE #27795: ZOOM Monitor 5 weird shifting behaviour
**Original reporter:** ISIS/Diego

### Expected behavior
Monitor 5 should shift correctly

### Actual behavior
Changing the shift in the GUI doesn't appear to do anything for monitor 5
Shifting by 1mm in a user file results in wildly different results

### Steps to reproduce the behavior
User file: *zoom*/masks/*191K_m5.txt*
Scatter: 7281
Transmission: 7281
Can Scatter / Transmission / Direct : 7280

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**

In a previous commit I added we treated 0. offset as a special case of
doing nothing, as naively I assumed the IDF position of M5 is correct

However, this skips the logic which repositions the monitor at the rear
detector, then applies a shift. So a shift in rear-detector relative to
the monitor IDF position would result in a different result.

This lead to confusing behaviour and incorrect results when a shift of 0
is set.

**Report to:** ISIS/Diego

**To test:**
- Open the ISIS SANS interface
- Use this 
[UserFile.txt](https://github.com/mantidproject/mantid/files/4118611/MaskFile.txt) as the UserFile
- In the user file check `TRANS/TRANSPEC=5 /shift=5` is present
- Load this batch file (you will need to rename to .csv) 
[repro.txt](https://github.com/mantidproject/mantid/files/4118616/repro.txt)
- Hit process all
- Go to convert units and convert - `sans_interface_raw_data -> 7281_trans_nxs` into wavelength
- Delete all other workspaces
- In the user file change the line to read `/shift=0`
- Reload the user file (you have to click the user file selection box again)
- Process again and convert to wavelength
- Compare spectrum 3/5, they should visually look similar (the gap previously was 1/2 the plot so it's obvious)

Fixes #27795 (there is another issue which will be fixed in #27791 )

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v4.3.0/sans.rst [hunk 1]
```diff
@@ -19,4 +19,9 @@ Improved
   can copy the history of a workspace to their clipboard or a file and the data
   will be reproduced on that machine without requiring editing of the script.
 
+Fixed
+#####
+- A zero monitor shift previously did not shift to account for the new position
```
Comment (giovannidisiena) [inline docs/source/release/v4.3.0/sans.rst]: Massive nitpick but could this be reordered to read better?
```suggestion
- A zero monitor shift did not previously shift to account for the new position
```

File: scripts/SANS/sans/algorithm_detail/move_workspaces.py [hunk 2]
```diff
@@ -103,11 +103,6 @@ def move_backstop_monitor(ws, move_info, monitor_offset, monitor_spectrum_number
     :param monitor_offset: Offset to shift by (m)
     :param monitor_spectrum_number: The spectrum number of the monitor to shift
     """
-    # Back out early so we don't shift the monitor from the IDF position to the
-    # rear detector position by accident
-    if monitor_offset == 0.0:
-        return
-
     monitor_spectrum_number_as_string = str(monitor_spectrum_number)
     # TODO we should pass the detector ID through, not the spec num
```
Comment (giovannidisiena) [inline scripts/SANS/sans/algorithm_detail/move_workspaces.py]: I know this isn't part of this PR but is there an issue to bring this in line with `Instrument 2.0`?

**Review by giovannidisiena — CHANGES_REQUESTED** [review]
Works well and looks good, just a couple of minor comments then I'll be happy to approve.

File: scripts/SANS/sans/algorithm_detail/move_workspaces.py (hunk 2 above)
Comment (DavidFair) [reply scripts/SANS/sans/algorithm_detail/move_workspaces.py]: Thanks for pointing it out anyway, we have #27024

**Review by giovannidisiena — DISMISSED** [review]
:shipit:

File: docs/source/release/v4.3.0/sans.rst [hunk 3]
```diff
@@ -19,4 +19,9 @@ Improved
   can copy the history of a workspace to their clipboard or a file and the data
   will be reproduced on that machine without requiring editing of the script.
 
+Fixed
+#####
+- A zero monitor shift did not previously shift to account for the position
```
Comment (gvardany) [inline docs/source/release/v4.3.0/sans.rst]: Can you be a bit more specific in the notes (which algorithm ? which instruments? in which context?)

**Comment by DavidFair** [comment]
@giovannidisiena / @gvardany would you be able to look at this again
Updating the release notes to clarify who this affects has dismissed the review

File: docs/source/release/v4.3.0/sans.rst [hunk 4]
```diff
@@ -19,4 +19,10 @@ Improved
...
```
Comment (giovannidisiena) [inline docs/source/release/v4.3.0/sans.rst]:
```suggestion
- A zero monitor shift did not previously account for the position
```

**Review by giovannidisiena — CHANGES_REQUESTED** [review]

File: docs/source/release/v4.3.0/sans.rst (hunk 4 above)
Comment (DavidFair) [reply docs/source/release/v4.3.0/sans.rst]: Good clarification

**Review by giovannidisiena — APPROVED** [review]

**Pull request merged**